api/cds.proto

```diff
@@ -26,17 +26,9 @@ service ClusterDiscoveryService {
   }
 }
 
-message ClusterDiscoveryRequest {
-  // The version_info provided in the request messages will be the version_info
-  // received with the most recent successfully processed response or empty on
-  // the first request.
-  bytes version_info = 1;
-  Node node = 2;
-}
-
 message ClusterDiscoveryResponse {
   bytes version_info = 1;
-  repeated Cluster clusters = 2;
+  repeated Cluster resources = 2;
```

Comment:
This is to allow some C++ evil in Envoy. I take the various xResponse objects in the Subscription client object as template parameters. Without knowing anything about their implementation details, I can populate the response resources when reading config from the filesystem (as opposed to gRPC/REST) by just accessing this field. It's basically making all the xResponse objects duck typed to have a `resources` mixin.